modules/aleph/includes/aleph.reservation.inc

```diff
@@ -213,8 +215,57 @@ function aleph_reservation_list($account, $requested_type = NULL) {
   return $result;
 }
 
-function aleph_reservation_create() {
-  return array();
+/**
+ * Create a new reservation in the library system.
+ *
+ * @param object $account
+ *   Drupal user account object.
+ * @param string $id
+ *   ID of reservation.
+ *
+ * @return array
+ *   Reservation information.
+ *
+ * @throws DingProviderReservationExists
+ * @throws DingProviderReservationNotAllowed
+ * @throws DingProviderReservationNotAvailable
+ */
+function aleph_reservation_create($account, $id) {
+  $patron_handler = new AlephPatronHandler(aleph_client());
+  $patron_handler->setPatron(aleph_get_patron_by_account());
+
+  // Check if the users has this reservation and throw exception.
+  $reservation_exists = FALSE;
+  $reservations = $patron_handler->getReservations();
+
+  foreach ($reservations as $reservation) {
+    if ($id === $reservation->getItem()->getId()) {
+      $reservation_exists = TRUE;
+    }
+  }
+
+  if ($reservation_exists) {
+    throw new DingProviderReservationExists();
+  }
+
```

Comment:
[I spotted some discussion about handling of reservations for items already on the shelf](https://reload.atlassian.net/secure/RapidBoard.jspa?rapidView=297&view=detail&selectedIssue=BBS-84). That does not seem to be handled here. Is that intentional?

Reply:
getReplyCode would return `0025` in that case.
`$response->success()` would return `FALSE` and an exception is thrown with the error message: "Item or like copy is on shelf. Cannot place hold request."